langstream-agents/langstream-agent-s3/src/main/java/ai/langstream/agents/s3/S3Processor.java

```diff
@@ -0,0 +1,194 @@
+import ai.langstream.ai.agents.commons.MutableRecord;
+import ai.langstream.api.runner.code.AbstractAgentCode;
+import ai.langstream.api.runner.code.AgentProcessor;
+import ai.langstream.api.runner.code.Header;
+import ai.langstream.api.runner.code.Record;
+import ai.langstream.api.runner.code.RecordSink;
+import ai.langstream.api.runtime.ComponentType;
+import com.samskivert.mustache.Mustache;
+import com.samskivert.mustache.Template;
+import io.minio.GetObjectArgs;
+import io.minio.GetObjectResponse;
+import io.minio.MinioClient;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class S3Processor extends AbstractAgentCode implements AgentProcessor {
+    private String bucketName;
+    private MinioClient minioClient;
+    private Template objectTemplate;
+
+    @Override
+    public void init(Map<String, Object> configuration) throws Exception {
+        bucketName = configuration.getOrDefault("bucketName", "langstream-source").toString();
+        String endpoint =
+                configuration
+                        .getOrDefault("endpoint", "http://minio-endpoint.-not-set:9090")
+                        .toString();
+        String username = configuration.getOrDefault("access-key", "minioadmin").toString();
+        String password = configuration.getOrDefault("secret-key", "minioadmin").toString();
+        String region = configuration.getOrDefault("region", "").toString();
+        String objectName = configuration.getOrDefault("objectName", "").toString();
+
+        // Object name is a mustache template because it is passed in the record
+        objectTemplate = Mustache.compiler().compile(objectName);
+
+        log.info(
+                "Connecting to S3 Bucket at {} in region {} with user {}",
+                endpoint,
+                region,
+                username);
+
+        MinioClient.Builder builder =
+                MinioClient.builder().endpoint(endpoint).credentials(username, password);
+        if (!region.isBlank()) {
+            builder.region(region);
+        }
+        minioClient = builder.build();
+    }
+
+    @Override
+    public void process(List<Record> records, RecordSink recordSink) {
+        for (Record record : records) {
+            processRecord(record, recordSink);
+        }
+    }
+
+    @Override
+    public ComponentType componentType() {
+        return ComponentType.PROCESSOR;
+    }
+
+    private void processRecord(Record record, RecordSink recordSink) {
+        log.debug("Processing record {}", record.toString());
+        MutableRecord context = MutableRecord.recordToMutableRecord(record, true);
+        final JsonRecord jsonRecord = context.toJsonRecord();
+
+        String fileName = objectTemplate.execute(jsonRecord);
+
+        log.info("Processing file {}", fileName);
```

Comment:
we should define what happens in case fileName is null or empty, maybe we  could "skip" the record ? (emit an empty list)